### PR TITLE
Filter scan notifications

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/scannotifications/messages/EventMessages.scala [hunk 1]
```diff
@@ -19,41 +19,55 @@ object EventMessages {
   case class SlackMessage(blocks: List[SlackBlock])
 
   implicit val scanEventMessages: Messages[ScanEvent] = new Messages[ScanEvent] {
+    private val allowedTags = Set("latest", "intg", "staging", "prod")
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/scannotifications/messages/EventMessages.scala:22]: Are we tagging any images with "mgmt"? If so do we need to account for that?

File: src/test/scala/uk/gov/nationalarchives/scannotifications/LambdaSpecUtils.scala [hunk 2]
```diff
@@ -19,9 +19,11 @@ class LambdaSpecUtils extends AnyFlatSpec with Matchers with BeforeAndAfterAll w
...
   }
 
-  val scanEvent1: ScanEvent = ScanEvent(ScanDetail("", ScanFindingCounts(Some(10), Some(100), Some(1000), Some(10000))))
-  val scanEvent2: ScanEvent = ScanEvent(ScanDetail("", ScanFindingCounts(Option.empty, Some(0), Option.empty, Some(10))))
-  val scanEvent3: ScanEvent = ScanEvent(ScanDetail("", ScanFindingCounts(Option.empty, Option.empty, Option.empty, Option.empty)))
+  val scanEvent1: ScanEvent = ScanEvent(ScanDetail("", List("latest"), ScanFindingCounts(Some(10), Some(100), Some(1000), Some(10000))))
+  val scanEvent2: ScanEvent = ScanEvent(ScanDetail("", List("latest"), ScanFindingCounts(Option.empty, Some(0), Option.empty, Some(10))))
+  val scanEvent3: ScanEvent = ScanEvent(ScanDetail("", List("latest"), ScanFindingCounts(Option.empty, Option.empty, Option.empty, Option.empty)))
+  val scanEvent4: ScanEvent = ScanEvent(ScanDetail("", List("anothertag"), ScanFindingCounts(Option.empty, Option.empty, Option.empty, Option.empty)))
+  val scanEvent5: ScanEvent = ScanEvent(ScanDetail("", List("latest"), ScanFindingCounts(Some(0), Some(0), Some(0), Some(0))))
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/scannotifications/LambdaSpecUtils.scala:26]: `scanEvent5` does not appear to be referenced anywhere. Is it being used?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of questions/comments

File: src/main/scala/uk/gov/nationalarchives/scannotifications/messages/EventMessages.scala (hunk 1 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/scannotifications/messages/EventMessages.scala]: We don't currently but we might in the future so it's worth adding now.

File: src/test/scala/uk/gov/nationalarchives/scannotifications/LambdaSpecUtils.scala (hunk 2 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/scannotifications/LambdaSpecUtils.scala]: I've added it into the test table. I had to change the tag to make it different from one of the other tests.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**